packages/next/next-server/server/image-optimizer.ts

```diff
@@ -86,7 +86,23 @@ export async function imageOptimizer(
       return { finished: true }
     }
 
-    if (!domains.includes(hrefParsed.hostname)) {
+    if (
+      !domains.some((domain) => {
+        if (domain.includes('*')) {
+          const domainExpr = new RegExp(
+            '^' +
+              domain
+                .split(/\*+/)
+                .map((s) => s.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&'))
+                .join('.*') +
```

Comment:
```suggestion
                .join('.+') +
```

I think this should match one or more, right?